practical/ProcessMining/group2/conformance_checking/src/comparison.py

```diff
@@ -0,0 +1,234 @@
+            The variant of the replay algorithm to be used. Default is variants.extensive.
+
+        Returns:
+        -------
+        tuple
+            A tuple containing the original FootPrintMatrix and the replayed FootPrintMatrix.
+        """
+        # Pipeline:
+        # 1. Get footprint matrix from log
+        fpm_original = FootPrintMatrix(input_log)
+        fpm_original.generate_footprint()
+        # 2. Run algorithm on log
+        net, start, end = get_model_from_pm4py(input_log, algorithm)
+        # 3. Replay resulting model from algorithm
+        logs_replayed = get_traces_with_replay(net, start, end, variant)
+        # 4. Get footprint matrix from replayed log
+        fpm_replayed = FootPrintMatrix(logs_replayed)
+        fpm_replayed.generate_footprint()
+        return fpm_original, fpm_replayed
+
+    def log_2_log(
+        self, event_log: str, algorithm: 'AlgoPm4Py'
+    ) -> List[Tuple[str, float]]:
+        """
+        This function takes a log and an algorithm, splits it into multiple sublogs.
+        Next, after running the algorithm on each sublog, the resulting model is being replayed in order to get a new log.
+        Finally, the resulting logs are being compared to the original log.
+
+
+        Parameters:
+        ----------
+        event_log: str
+            Path to the event log.
+        algorithm: AlgoPm4Py
+            Algorithm to be used.
+
+        Returns:
+        -------
+        List of tuples, where each tuple contains the name of the sublog and the conformance value.
+        """
+
+        def split_log_into_sublogs(log):
+            sublogs = []
+            # Initial split to divide the log into two equal parts
+            split_1, split_2 = split_train_test.split(log, train_percentage=0.5)
+
+            # Further split each part into two sublogs
+            sublog_1, sublog_2 = split_train_test.split(split_1, train_percentage=0.5)
+            sublog_3, sublog_4 = split_train_test.split(split_2, train_percentage=0.5)
+
+            sublogs.extend([sublog_1, sublog_2, sublog_3, sublog_4])
+            return sublogs
+
+        log = get_log_from_file(event_log)
+
+        # Split log into sublogs. We use 4 sublogs for ease of use.
+        sublogs = split_log_into_sublogs(log)
+
+        replayed_logs = []
+        # Run algorithm on each sublog
+        for i, sublog in enumerate(sublogs):
+            # Run pipeline
+            footprint_of_log, footprint_of_replayed_log = self.pipeline(
+                sublog, algorithm
+            )
+            visualize_sorted_dict(
+                footprint_of_replayed_log.relations, "l2l_{}".format(i)
+            )
+            replayed_logs.append(footprint_of_replayed_log)
+
+        # Compare with original log
+        footprint_of_log, x = self.pipeline(log, algorithm)
+        visualize_sorted_dict(footprint_of_log.relations, "l2l_original")
+        comparison_values = []
+        conformance_checking = ConformanceChecking()
```

Comment:
Arguable, if conformance checking needs it's own object that needs to be initialized instead of two functions `conformance.get_value` and `conformance.get_matrix`, object init is actually redundant.